src/shared/modules/user-favorites/client.test.js

```diff
@@ -0,0 +1,190 @@
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import { assign, head, map, slice } from 'lodash-es'
+
+import {
+  createManyUserFavorites,
+  createUserFavorite,
+  getAllUserFavorites,
+  removeManyUserFavorites,
+  removeUserFavorite,
+  updateManyUserFavorites,
+  updateUserFavorite
+} from './client'
+
+import { reset } from './user-favorites.utils'
+
+jest.mock('uuid/v4', () => () => 'apa')
+jest.mock('./user-favorites.utils', () => {
+  const mockPrefix = '/foo/'
+  const favs = [
+    { path: mockPrefix, contents: 'bar', id: '1' },
+    { path: `${mockPrefix}baz`, contents: 'bam', id: '2' }
+  ]
+  let mockFavorites = favs
+  return {
+    reset () {
+      mockFavorites = favs
+    },
+    tryGetUserFavoritesLocalState () {
+      return mockFavorites
+    },
+    setUserFavoritesLocalState (updated) {
+      mockFavorites = updated
+    }
+  }
+})
+
+const mockPrefixCopy = '/foo/'
+const mockFavoritesCopy = [
+  { path: mockPrefixCopy, contents: 'bar', id: '1' },
+  { path: `${mockPrefixCopy}baz`, contents: 'bam', id: '2' }
+]
+
+describe('user-favorites/client', () => {
+  beforeEach(() => {
+    reset()
+  })
+
+  describe('createUserFavorite', () => {
+    test('should be unary', () => {
+      expect(createUserFavorite.length).toBe(1)
+    })
+    test('should add one favorite to localstate', async () => {
+      const data = { contents: 'bom' }
+      const expected = { contents: 'bom', id: 'apa' }
+      const newState = [...mockFavoritesCopy, expected]
+
+      await expect(createUserFavorite(data)).resolves.toEqual(expected)
+      await expect(getAllUserFavorites()).resolves.toEqual(newState)
+    })
+  })
+
+  describe('createManyUserFavorites', () => {
+    test('should be unary', () => {
+      expect(createManyUserFavorites.length).toBe(1)
+    })
+
+    test('should add one favorite to localstate', async () => {
+      const data = [{ contents: 'bom' }]
+      const expected = [{ contents: 'bom', id: 'apa' }]
+      const newState = [...mockFavoritesCopy, ...expected]
+
+      await expect(createManyUserFavorites(data)).resolves.toEqual(expected)
+      await expect(getAllUserFavorites()).resolves.toEqual(newState)
+    })
+
+    test('should add several favorites to localstate', async () => {
+      const data = [{ contents: 'bom' }, { contents: 'apa' }]
+      const expected = [
+        { contents: 'bom', id: 'apa' },
+        { contents: 'apa', id: 'apa' }
```

Comment:
It looks kind of strange with two favorites having the same id here in the tests since the whole point of `uuid` is to generate unique id's.
Rather than mocking `uuid`, could you change the assertion to be something like `expect.any(String)` or similar.

Not a big deal, just a bit unexpected.